### ISSUE TITLE
In recycleView  can not work well

### ISSUE THREAD

**Issue by haonan1** [Manabu-GT/ExpandableTextView#44]
In a RecyclerView i have other view, if the TextView is expanded and you scroll，the other view in some item has gone  please give me some suggestion

**Comment by jymycc** [comment]
have you fixed that ? i got the same problem: (

**Comment by kukreja-vikramaditya** [comment]
Yes same here. Can you create a expand() and collapse() method?

**Comment by mac8878** [comment]
@kukreja-vikramaditya @jymycc @haonan1 seems like setting the visibility gone and then Visible on `OnBindViewHolder` will fix it , (working  for me)

**Comment by HolenZhou** [comment]
same here. How to solve it?

**Comment by Manabu-GT** [comment]
If anyone can share sample code which reproduces what you are experiencing, I can take a look this weekend.

Thank you.

**Comment by kukreja-vikramaditya** [comment]
I don't think it is a problem with library, could you guys check if your 'position' or 'holder' has the keyword 'final' before it?

**Comment by Studentessa** [comment]
hello every body! I'm getting the same problem, I have tried using @billionbucks trick but it doesn't work. I'm not using final variables @kukreja-vikramaditya !
tanks in advance!

**Comment by 469277027** [comment]
@Studentessa Thanks,it's works very good

**Comment by deeps2** [comment]
@Studentessa thanks, it works.
 In```
 onBindViewHolder(ViewHolder holder, final int pos) {
      ....
       SparseBooleanArray mTogglePositions = new SparseBooleanArray();
       
      // setText(@Nullable CharSequence text, @NonNull SparseBooleanArray collapsedStatus, int pos)
       holder.newsDescription.setText(listOfArticles.get(position).getDescription(), mTogglePositions , pos); 
      //where newsDescription is my TextView
   
     ...
}
```

**Comment by nangsan** [comment]
@deeps2 your solution was right, but it has some flaw initialize the SparseBooleanArray while creating RecyclerView adapter and pass it to the setText method. That should do it.

**Comment by monikagupta94** [comment]
@Studentessa Thank you. It worked but also need to change animation 0  if content is enough long.

**Comment by BzCoder** [comment]
@Studentessa thanks a lot,you solve my problem!

**Comment by aiynmm** [comment]
@Studentessa What's your mean about what you said that need to change animation 0 if content is enough long? I have a problem when collaping the text if content is enough long!

**Comment by Studentessa** [comment]
@aiynmm  what your'asking was refered by @monikagupta94 , maybe he can answer you. =D

**Comment by aiynmm** [comment]
> @aiynmm what your'asking was refered by @monikagupta94 , maybe he can answer you. =D

I'm sorry, I misread it! @Studentessa

**Comment by aiynmm** [comment]
> @Studentessa Thank you. It worked but also need to change animation 0 if content is enough long.

@monikagupta94 Could you tell me what should I do to solve this problem!

**Comment by dongxingrong** [comment]
Solution: while using in recyclerview, use SpareIntArray to record mMarginBetweenTxtAndBottom and mCollapsedHeight, and reuse them when ExpandableTextView's init value is 0.
Reuse items in reycleview cause this problem.
